### PR TITLE
Bucket Sort in Java

### PR DESCRIPTION
**Pull request by ritvij14**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description

(Write your answer here.)
This is the code for the Bucket Sort algorithm in Java. In this algorithm, we are given a set of numbers which is uniformly distributed in a range. We use this range to divide the numbers into buckets of fixed size. Then we individually sort each of the buckets, which are then concatenated into a single array.
The best-case time complexity of the algorithm: O(n)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Issue #426 
(Write your answer here.)

### CODE REVIEW

**Comment by ritvij14** [comment]
@siddharth25pandey @HarshCasper please review and suggest changes. Will do them ASAP.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@ritvij14 Please make following modifications
* Instead of taking a random array, Take input from the user
* change the datatype of array to float for better visualization.
* Also modify the user-interface of the program for user-friendly experience by displaying the message 
   before & after the insertion of array from the user.

**Comment by ritvij14** [comment]
@siddharth25pandey I had a doubt related to the algorithm, how many buckets should I use? Is it ok to use 5 like I am doing right now? Or should I ask the user to input that as well?

**Comment by siddharth25pandey** [comment]
> @siddharth25pandey I had a doubt related to the algorithm, how many buckets should I use? Is it ok to use 5 like I am doing right now? Or should I ask the user to input that as well?

@ritvij14 it will be better if you take input from the user.

**Comment by ritvij14** [comment]
@siddharth25pandey done.

**Review by siddharth25pandey — APPROVED** [review]
@ritvij14 Looks fine to me
Able to merge.
Thanks!

**Comment by ritvij14** [comment]
@siddharth25pandey thank you. I had one doubt, should I start working on a new issue on another branch or wait for this to get merged? When will it be merged?

**Comment by siddharth25pandey** [comment]
> @siddharth25pandey thank you. I had one doubt, should I start working on a new issue on another branch or wait for this to get merged? When will it be merged?

@ritvij14 you can start working on another issue & your PR will be merged after **two** more mentors will give successful approval.

File: Java/sort/BucketSort.java
Comment (HarshCasper) [inline Java/sort/BucketSort.java]: Add a Comment block showing the sample Input/Output in this case. It would be ready to merge thereafter.

**Review by HarshCasper — COMMENTED** [review]
Kindly resolve this and it would be ready to merge.

File: Java/sort/BucketSort.java
Comment (ritvij14) [reply Java/sort/BucketSort.java]: Ok, it can be anyone of the sample Input/Output cases that I tried locally to test it out, right?

Comment (ritvij14) [reply Java/sort/BucketSort.java]: @HarshCasper Is this fine?

**Review by iamrajiv — APPROVED** [review]

**Review by iamrajiv — APPROVED** [review]

File: Java/sort/BucketSort.java
Comment (Kajol-Kumari) [inline Java/sort/BucketSort.java]: please add one extra blank line here.

**Review by Kajol-Kumari — APPROVED** [review]
@ritvij14 minor change required rest LGTM :)

File: Java/sort/BucketSort.java
Comment (ritvij14) [reply Java/sort/BucketSort.java]: @Kajol-Kumari done.

**Comment by ritvij14** [comment]
@iamrajiv I have done the changes. Please merge this.

**Pull request merged**
